internal_model_tool/common.py

```diff
@@ -0,0 +1,79 @@
+import boto3
+import base64
+import json
+import re
+from urllib.request import Request, urlopen
+
+
+def get_session(*, role_arn):
+    sts_client = boto3.client("sts")
+    assumed_role_object = sts_client.assume_role(
+        RoleArn=role_arn, RoleSessionName="AssumeRoleSession1"
+    )
+    credentials = assumed_role_object["Credentials"]
+    return boto3.Session(
+        aws_access_key_id=credentials["AccessKeyId"],
+        aws_secret_access_key=credentials["SecretAccessKey"],
+        aws_session_token=credentials["SessionToken"],
+    )
+
+
+def get_local_date():
+    config_file = open(
+        "../common/display/src/main/scala/weco/catalogue/display_model/ElasticConfig.scala",
+        "r",
+    )
+    config_text = config_file.read()
+    config_file.close()
+    date = re.findall('val indexDate = "(.*)"', config_text)[0]
+    return date
+
+
+def get_remote_latest_internal_model(session, date):
+    # this comes in the format of
+    # {"model.versions.4988":"884bf55a45d76171db88112142b2ee8dd6e37f46","model.versions.4960":"55842c7dffef3b80fe7755768728028d75f0d866"}
+    # So we choose the highest available version
+    meta = get_remote_meta(session, date)
+    version = max([int(key.split(".")[-1]) for key in meta])
+    hash = meta.get(f"model.versions.{version}")
+    return f"{version}.{hash}"
+
+
+def get_remote_meta(session, date):
+    sm = session.client("secretsmanager")
+    host = sm.get_secret_value(SecretId="elasticsearch/catalogue_api/public_host")[
+        "SecretString"
+    ]
+    username = sm.get_secret_value(
```

Comment:
You may want to create a user specifically for this tool (as with the diff_tool).